### ISSUE TITLE
Unable to use onprogress event

### ISSUE THREAD

**Issue by Connormiha** [albburtsev/talaria#7]
`onprogress` handler should be attached before `xhr.send()`.

If i will try:

``` js
let {xhr. promise} = api.postRequest(url, data, {headers: {/***/}});
xhr.addEventListener('progress', () => {/** will run only once in the end*/})
```

`progress` doesn't work if was atached after `xhr.send`. So i need to add the ability to attache `onProgress` callback.

What do you think aboud this? We can use 3th param.

``` js
api.postRequest(url, data, {headers: {/***/}, onProgress: (e) => {console.log(e)}})
```

**Comment by albburtsev** [comment]
Thanks! I'll add `onProgress` callback in `settings` soon.

**Comment by albburtsev** [comment]
I've released v0.3.0 with requested changes.
